haystack/nodes/file_converter/pdf.py

```diff
@@ -227,13 +284,18 @@ def _read_pdf(
         if not multiprocessing:
             for i in range(start_page, end_page):
                 page = doc[i]
-                document += page.get_text("text", sort=sort_by_position) + "\f"
+                partial_tp = None
+                if ocr is not None:
+                    partial_tp = page.get_textpage_ocr(
+                        flags=0, full=True if ocr == "full" else False, dpi=300, language=ocr_language
+                    )
```

Comment:
```suggestion
                if ocr is not None:
                    full = ocr == "full"
                    partial_tp = page.get_textpage_ocr(
                        flags=0, full=full, dpi=300, language=ocr_language
                    )
```

Am not sure whether we also need to also check if `ocr` is either `auto` or `full`. 🤔